code/game/objects/structures/false_walls.dm

```diff
@@ -87,10 +91,7 @@
 
 /obj/structure/falsewall/proc/ChangeToWall(delete = 1)
 	var/turf/T = get_turf(src)
-	if(!walltype || walltype == "metal")
-		T.ChangeTurf(/turf/closed/wall)
-	else
-		T.ChangeTurf(text2path("/turf/closed/wall/mineral/[walltype]"))
+	T.ChangeTurf(walltype)
```

Comment:
Remove the wall var because it was probably being used to pass their type onto falsewalls.